src/System.Private.CoreLib/shared/System/SpanHelpers.Byte.cs

```diff
@@ -1324,18 +1324,131 @@ public static unsafe int SequenceCompareTo(ref byte first, int firstLength, ref
             IntPtr offset = (IntPtr)0; // Use IntPtr for arithmetic to avoid unnecessary 64->32->64 truncations
             IntPtr nLength = (IntPtr)(void*)minLength;
 
-            if (Vector.IsHardwareAccelerated && (byte*)nLength > (byte*)Vector<byte>.Count)
+            if (Avx2.IsSupported)
             {
-                nLength -= Vector<byte>.Count;
-                while ((byte*)nLength > (byte*)offset)
+                if ((byte*)nLength >= (byte*)Vector256<byte>.Count)
                 {
-                    if (LoadVector(ref first, offset) != LoadVector(ref second, offset))
+                    nLength -= Vector256<byte>.Count;
+                    int matches;
+                    while ((byte*)nLength > (byte*)offset)
                     {
-                        goto NotEqual;
+                        matches = Avx2.MoveMask(Avx2.CompareEqual(LoadVector256(ref first, offset), LoadVector256(ref second, offset)));
+                        if (matches == -1)
+                        {
+                            // All matched
+                            offset += Vector256<byte>.Count;
+                            continue;
```

Comment:
Shall we remove this `continue` and use `else` instead? That would be consistent with the code below.

Reply:
Code below is in an `if` as we know it can only execute once (as double its length would be picked up by this code) whereas this one is in a `while`